generator/views/lookml_utils.py

```diff
@@ -117,23 +117,25 @@ def _generate_dimensions(client: bigquery.Client, table: str) -> List[Dict[str,
     """
     dimensions = {}
     for dimension in _generate_dimensions_helper(client.get_table(table).schema):
-        name = dimension["name"]
+        name_key = dimension["name"]
+
+        # This prevents `time` dimension groups from overwriting other dimensions below
+        if dimension.get("type") == "time":
+            name_key += "_time"
+
         # overwrite duplicate "submission", "end", "start" dimension group, thus picking the
         # last value sorted by field name, which is submission_timestamp
         # See also https://github.com/mozilla/lookml-generator/issues/471
         if (
-            name in dimensions
-            and name != "submission"
-            and not name.endswith("end")
-            and not name.endswith("start")
-            and not (name == "event" and dimension["type"] == "time")
-            # workaround for `mozdata.firefox_desktop.desktop_installs`
-            and not (name == "attribution_dltoken" and dimension["type"] == "time")
+            name_key in dimensions
+            and dimension["name"] != "submission"
+            and not dimension["name"].endswith("end")
+            and not dimension["name"].endswith("start")
```

Comment:
I'd be inclined to have these exceptions more explicitly targeted at dimension groups, as the relatively broad `endswith("end")` and `endswith("start")` conditions make me a bit nervous.  Something like
```suggestion
            and name_key != "submission_time"
            and not name_key.endswith("end_time")
            and not name_key.endswith("start_time")
```
or
```suggestion
            and not (
                dimension.get("type") == "time"
                and (
                    dimension["name"] == "submission" 
                    or dimension["name"].endswith("end")
                    or dimension["name"].endswith("start")
                )
            )
```

Reply:
Yeah, this part is confusing, narrowing this check down won't hurt.